pkg/apis/core/types.go

```diff
@@ -4383,8 +4383,20 @@ type ConfigMap struct {
 
 	// Data contains the configuration data.
 	// Each key must consist of alphanumeric characters, '-', '_' or '.'.
+	// Values with non-UTF-8 byte sequences must use the BinaryData field.
+	// The keys stored in Data must not overlap with the keys in
+	// the BinaryData field.
```

Comment:
What happens if the keys collide.  Spec it.  will it be rejected by validation?  Will one be dropped?  will it crash the server?